doc/query.md

```diff
@@ -74,95 +69,80 @@ Basic querying of fields benefits but does _not require_ a JSON index. For examp
 
-The first argument, `fieldNames` is a list of fields to put into the index. The second argument, `indexName` is a name for the index. This is used to delete indexes at a later stage and appears when you list the indexes in the database.  The third argument, `indexType` defines what type of index to create. If not provided, the index type defaults to `JSON`.  The fourth argument, `indexSettings` is comprised of index parameters and their values.  Currently the only valid index setting is `tokenize` and it can only apply to a TEXT index.  If index settings are not provided for a TEXT index, the `tokenize` parameter defaults to the value `simple`.
+The first argument, `fields` is a list of fields to put into the index. The second argument, `indexName` is a name for the index. This is used to delete indexes at a later stage and appears when you list the indexes in the database; `indexName` is optional - if `null` is specified then a name will be generated. For TEXT indexes, `tokenizer` specifies the SQLite FTS tokenizer to use.
 
 A field can appear in more than one index. The query engine will select an appropriate index to use for a given query. However, the more indexes you have, the more disk space they will use and the greater the overhead in keeping them up to date.
 
 To index values in sub-documents, use _dotted notation_. This notation puts the field names in the path to a particular value into a single string, separated by dots. Therefore, to index the `species` field of the `pet` sub-document in the examples above, use `pet.species`.
 
 ```java
 // Create an index over the name, age, and species fields.
-String name = im.ensureIndexed(Arrays.<Object>asList("name", "age", "pet.species"),
-                               "basic");
-if (name == null) {
+try {
+    Index i = q.createJsonIndex(Arrays.<FieldSort>asList(new FieldSort("name"), new FieldSort("age"), new FieldSort("pet.species")), "basic");
+} catch (QueryException) {
     // there was an error creating the index
 }
 ```
 
 ####Indexing for text search
 
-Since text search relies on SQLite FTS which is a compile time option, we must ensure that SQLite FTS is infact available.  To verify that text search is enabled and that a text index can be created use `isTextSearchEnabled()` before attempting to create a text index.  If text search is not enabled see [compiling and enabling SQLite FTS][enableFTS] and [SQLite Android Bindings][androidBind] for details.
+Since text search relies on SQLite FTS which is a compile time option, we must ensure that SQLite FTS is available.  To verify that text search is enabled and that a text index can be created use `isTextSearchEnabled()` before attempting to create a text index.  If text search is not enabled see [compiling and enabling SQLite FTS][enableFTS] for details.
 
 [enableFTS]: http://www.sqlite.org/fts3.html#section_2
-[androidBind]: https://www.sqlite.org/android/doc/trunk/www/index.wiki
 
 ```java
 if (im.isTextSearchEnabled()) {
```

Comment:
Thanks for spotting. I've also fixed a few more code errors which I caught by copying and pasting the code blocks into Android Studio, so hopefully everything's done now.